### ISSUE TITLE
Maximum string length not correctly checked when transforming to XDR

### ISSUE THREAD

**Issue by TorstenStueber** [stellar/js-xdr#33]
**Describe the bug**
The XDR standard RFC 4506 defined strings to be ASCII strings only. In this package strings are allowed to use full unicode character set and use UTF-8 encoding.
This extension is not a problem – however, the maximum length of a string when transforming to XDR is not correctly checked. The maximum string length is defined to refer to the maximum number of bytes but here the length of the string is used instead (which can be much shorter).

**What version are you on?**
1.1.2

**To Reproduce**
1. npm init -y
2. npm install js-xdr
3. Run the following code
```js
const XDR = require("js-xdr");

const types = XDR.config(xdr => {
  xdr.struct("A", [["b", xdr.string(32)]]);
});

console.log(new types.A({ b: "€€€€€€€€€€€€€€€€€€€€€€€€€€€€€€€€" }).toXDR());
```

**Expected behavior**
The code returns an error as the string will take 96 bytes (way more than the maximum of 32 bytes).

**Actual behavior**
The code runs fine as the length of the string is 32.

**Additional context**
Strings with maximum length are used plentiful in the XDR type definitions of Stellar. I suppose that the xdr decoder in Stellar core (or Stellar horizon) will not accept the above XDR and fail – as expected. But then also this javascript decoder should already fail.

**Comment by ire-and-curses** [comment]
Related: [go-xdr:1](https://github.com/stellar/go-xdr/issues/1)

**Comment by Shaptic** [comment]
Closed by #91.
